[IMP] pos/viva: Update Viva.com doc (#13848)
odoo/documentation: content/applications/sales/point_of_sale/payment_methods/terminals/viva_com.rst

Viva.com
========

**Viva.com** is a payment service that offers payment solutions through the **Viva Terminal** app
for :doc:`physical <../terminals>` and virtual terminals.

.. important::
   - Viva.com payment terminals do not require an :doc:`IoT Box </applications/general/iot>` to
     operate.
   - `Many European countries <https://developer.viva.com/about-viva/>`_ support the use of Viva.com
     payment terminals.
   - The Viva Terminal app turns a smartphone with an NFC chip into a `payment terminal
     <https://www.viva.com/en-gb/blog/tap-on-phone-turn-your-phone-into-a-mobile-card-reader>`_.
   - Odoo only supports the :icon:`fa-eur` (:guilabel:`Euro`) currency with Viva.com.

.. seealso::
   - `List of supported terminals <https://www.vivawallet.com/shop/terminals/terminals-4>`_.
   - `List of supported payment methods <https://developer.viva.com/payment-methods/>`_.

.. _viva/configuration:

Viva.com configuration
======================

To configure a Viva terminal, go to the `Viva.com website <https://www.viva.com>`_, create an

> account, and then follow these steps:

Author: Nice suggestion, I've updated the first step that explains how to get to the API Access screen. This allows users to avoid clicking the  first two links and simply do the steps directly from our doc: 

#. On the viva.com dashboard, go to :menuselection:`Settings --> API Access --> General`.
#. Copy the `Merchant ID and API key
   <https://developer.viva.com/getting-started/find-your-account-credentials/merchant-id-and-api-key/>`_.
#. Copy the `Client ID and the generated Client secret (POS API credentials)
   <https://developer.viva.com/getting-started/find-your-account-credentials/pos-apis-credentials/>`_.

#. Copy the `Merchant ID and API key
   <https://developer.viva.com/getting-started/find-your-account-credentials/merchant-id-and-api-key/>`_.
#. Copy the `Client ID and generate the Client secret (POS API credentials)
   <https://developer.viva.com/getting-started/find-your-account-credentials/pos-apis-credentials/>`_.
#. Download the Viva Terminal app on a device, then generate and copy the `activation code
   <https://euhelp.viva.com/en/articles/5316775-how-do-i-activate-the-viva-com-terminal>`_.
#. Back to the Viva website dashboard, `create a new card terminal, and paste the activation code
   <https://euhelp.viva.com/en/articles/5316775-how-do-i-activate-the-viva-com-terminal>`_.
#. Copy the :guilabel:`Terminal ID` generated by the terminal activation.

.. warning::
   The POS API credentials are only displayed once. Make sure to keep a copy to secure them.

.. note::
   - The POS API credentials are for APIs that use Basic Authentication, including those for Android
     and iOS POS activation and the Cloud Terminal API.

Odoo POS configuration
======================

To connect the Viva terminal with Odoo Point of Sale, follow these steps:

> #. Go to :menuselection:`Configuration --> Settings --> Payment Terminals`, enable the
>    :guilabel:`Viva Wallet` terminal, and click :guilabel:`Save`.
> #. Go to :menuselection:`Configuration --> Payment Methods --> New` and :doc:`create a payment
>    method <../../payment_methods>`.

Reviewer: - Don't forget the app name in menuselections
- Only menu items in menuselections; the rest should use guilabels

```suggestion
#. Go to :menuselection:`Point of Sale --> Configuration --> Settings`, scroll down to the :guilabel:`--> Payment Terminals`, enable the :guilabel:`Viva Wallet` terminal, and click :guilabel:`Save`.
#. Go to :menuselection:`Point of Sale --> Configuration --> Payment Methods` and :doc:`create a payment
   method <../../payment_methods>`.
```
(since the whole payment method creation process is explained on that doc page)

> #. Select :guilabel:`Terminal` in the :guilabel:`Integration` field.

Reviewer: Or 
```suggestion
#. Set the :guilabel:`Integration` field to guilabel:`Terminal` and select the desired :guilabel:`Journal`.
```
- 1st change is optional but that way we make it easier for people to find the field since it's the first item mentionned
- 2nd one, as discussed; if we don't select a journal we don't see the Integrate with field and we can't proceed with the rest of the steps

#. Select :guilabel:`Viva Wallet` in the :guilabel:`Integrate with` field.
#. Paste the copied information from :ref:`Viva.com <viva/configuration>` into the corresponding
   fields:

   - :guilabel:`Merchant ID`
   - :guilabel:`API Key`
   - :guilabel:`Client ID`
   - :guilabel:`Client secret`
   - :guilabel:`Terminal ID`
#. Save the form and copy the generated webhook URL from the :guilabel:`Viva Wallet Webhook
   Endpoint` field.
#. Go to the :ref:`Viva.com <viva/configuration>` account and paste the webhook URL into the
   `corresponding field
   <https://developer.viva.com/webhooks-for-payments/transaction-payment-created/>`_.

> #. Back to Odoo, go to :menuselection:`Configuration --> Settings --> Payment --> Payment Methods`,
>    and add the created payment method.

Reviewer: ```suggestion
#. In Odoo, :ref:`access the POS settings <configuration/settings>, scroll down to the :guilabel:`Payment` section, and add the created payment method in the :guilabel:`Payment mthods` field.
```
This field is specific to the selected POS in the settings I think so I suggest referring to the doc page where that's explained (like I did in my suggestion) or saying you need to go to the settings, select the POS, then add the method (not in that exact way of course but you get the idea :wink: )

Viva.com
========

Connecting a **Viva.com** :doc:`payment terminal <../terminals>` gives customers a fluid payment

> flow and eases cashiers' work.

Author: Hey @auva-odoo, I linked some steps to the Viva.com documentation. I don't think it's relevant for us to describe their flow while they have their own documentation for it. 

After discussing with BAs and Devs, the Viva Terminal app works with both physical and virtual terminals, and the configuration is the same for both types. I hope the one intro line clarifies this now ;)

.. important::
   - Viva.com payment terminals do not require an :doc:`IoT Box </applications/general/iot>` to
     operate.
   - Many European countries support the use of Viva.com payment terminals. Check the `availability
     of the Viva Terminal app in Europe <https://developer.viva.com/about-viva/>`_.

.. note::
   Viva.com turns phones into card readers with `tapping options
   <https://www.viva.com/en-gb/blog/tap-on-phone-turn-your-phone-into-a-mobile-card-reader>`_.

> Configuration

Author: @auva-odoo , I tried to simplify the Viva.com configuration as much as possible to avoid writing steps that are already explained in the Viva.com documentation. However, for payment terminals configuration, there will often be a back-and-forth setup between the terminal website and Odoo.
